docs/content/use-cases/write-heavy-volumes.md

```diff
@@ -0,0 +1,93 @@
+of mounts available with Docksal, however any non-Linux host machine will take a performance hit.
+
+There are a few options for boosting the speed of your application available, as noted in the [Docksal Volumes](/core/volumes) section,
+but another potential method of boosting performance is to keep write-heavy volumes inside the containers so that Docksal does not
+need to rely on mounts to the host to read or write data.
+
+The process of keeping write-heavy volumes inside containers involves mounting certain folders as volumes within the CLI container and
+interacting with the data directly through the container. For example, in a Composer-based Drupal project, the `/vendor` folder is usually pretty large
+and includes a lot of functionality and code that Drupal needs to run. In a standard Docksal application, the process for interacting with these files is
+similar to this:
+```
+macOS Host ==============> Docker Desktop ==============> Container
+(~/project/vendor)                                        (/var/www/vendor)
+```
+
+If you mount the write-heavy folder as a volume directly inside a container, the container's filesystem can interact with it at native filesystem speeds, because the files exist within the container instead of mounted from the host.
+
+The downside to this is that changes to the files in these folders on the host are not reflected in the container and vice versa, however a command can be written
+to sync between the host and container.
+
+A basic example of this would be to mount the `vendor` and the `docroot/modules/contrib` folders in a standard, Composer-based Drupal project.
+
+In your `docksal.yml` file, you would have something like this:
+
+```yaml
+version: 3.9
+services:
+  cli:
+    volumes:
+      # Mount the vendor and contrib folders to volumes within the container.
+      - drupal_vendor:/var/www/vendor
+      - drupal_contrib:/var/www/docroot/modules/contrib
+volumes:
+  # Define the volumes.
+  drupal_vendor:
+  drupal_contrib:
+```
+
+There is a more verbose example of this type of customization in the [Drupal + GatsbyJS Boilerplate docksal.yml](https://github.com/docksal/boilerplate-drupal-gatsby/blob/master/.docksal/docksal.yml).
+
+{{% notice note %}}
+There does not need to be anything after the `:` in the `volumes:` section of your `docksal.yml`. This defines the named volumes. No further settings are required.
+{{% /notice %}}
+
+Now when you spin up your project, instead of mounting these folders from the host machine, separate, empty volumes will be created
+and mounted at the defined locations within your container.
+
+Because these volumes are only accessible within the container, any commands that interact with them _MUST_ be run within the container. For example,
+if you normally run `composer install` outside of the container and rely on the bind mount to link the host data to the container data, you will need
```

Comment:
Note: Running  `composer install` directly on the host is not a good practice. I can see why users would be doing so, but it should be highly discouraged.